plugin/src/main/java/org/opensearch/ml/utils/MLNodeUtils.java

```diff
@@ -109,10 +109,14 @@ public static String processRemoteInferenceInputDataSetParametersValue(String in
                 if (value.isTextual()) {
                     String textValue = value.asText();
                     try {
-                        // Try to parse the string as JSON
-                        JsonNode parsedValue = mapper.readTree(textValue);
-                        // If successful, replace the string with the parsed JSON
-                        parametersNode.set(key, parsedValue);
+                        // Only try to parse as JSON if it looks like a JSON object or array
+                        if ((textValue.startsWith("{") && textValue.endsWith("}"))
```

Comment:
What about the else branch? In some cases, the previous logic is to catch the exception and set the value. But now nothing will happen.

Reply:
The intent of doing nothing is keeping value exactly as is from line 107.

But as other reviewers have noted, this breaks the intent of this method so is likely the wrong fix.  Leaving here in draft for discussion purposes.

Reply:
Oops. Fixed.